modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -1732,23 +1879,102 @@ public function massageFormValues(array $values, array $form, FormStateInterface
+      if (!empty($value['tabs']['advanced']['display']['options']['enableResultsCountText']) && $value['tabs']['advanced']['display']['options']['enableResultsCountText']) {
+        $config['interface']['display']['options']['resultsCountText'] = $value['tabs']['advanced']['display']['options']['resultsCountText'] ?? '';
+      }
       $config['interface']['display']['options']['noResultsText'] = $value['tabs']['advanced']['display']['options']['noResultsText'] ?? '';
       $config['interface']['display']['options']['loadingText'] = $value['tabs']['advanced']['display']['options']['loadingText'] ?? '';
       $config['interface']['display']['options']['errorText'] = $value['tabs']['advanced']['display']['options']['errorText'] ?? '';
+      // To enable pagination no ui feature yet.
+      $config['interface']['display']['options']['pagination'] = ['type' => 'numbers'];
 
       $value['value'] = json_encode($config);
+      $errors = $this->validateJson($value['value']);
+      if (!empty($errors)) {
+        $field_name = $this->fieldDefinition->getName();
+        $form_state->setErrorByName($field_name, $this->t('JSON does not validate against the schema. Violations: @errors.', [
+          '@errors' => implode(' - ', $errors),
+        ]));
+      }
     }
     return $values;
   }
 
+  /**
+   * Callback to validate the JSON.
+   *
+   * @param string $json
+   *   The json string value.
+   *
+   * @return array
+   *   The error array list if exists.
+   */
+  public function validateJson(string $json) {
+    $errors = [];
+    if (!empty($json)) {
+      $json_object = json_decode($json);
+      if ($json_object === NULL) {
+        $errors[] = $this->t('Invalid JSON.');
+      }
+      else {
+        // Validate against the JSON Schema.
+        $json_schema = $this->fieldDefinition->getFieldStorageDefinition()->getSetting('schema');
+        if (!empty($json_schema)) {
+          $json_schema_object = json_decode($json_schema);
+          $schema_storage = new SchemaStorage();
+          $schema_storage->addSchema('file://content_collection_configuration_schema', $json_schema_object);
+          $json_validator = new Validator(new Factory($schema_storage));
+          $num_errors = $json_validator->validate($json_object, $json_schema_object);
+          if ($num_errors) {
+            foreach ($json_validator->getErrors() as $error) {
+              $errors[] = $this->t('[@property] @message', [
+                '@property' => $error['property'],
+                '@message' => $error['message'],
+              ]);
+            }
+          }
+        }
+      }
+    }
+    return $errors;
+  }
+
+  /**
+   * Get all index entity reference fields along with name value fields.
+   *
+   * @return array|null
+   *   The entity reference fields array, or NULL if the none have a name value.
+   */
+  public function getValidatedIndexEntityReferenceFields() : array {
```

Comment:
@joshua-salsadigital fatal error will occur when a NULL value is returned.
```suggestion
  public function getValidatedIndexEntityReferenceFields() : ?array {
```

Reply:
does it need to be public?

Reply:
Updated, Made protected.